### ISSUE TITLE
Make preview generation configurable

### ISSUE THREAD

**Issue by drikqlis** [nextcloud/richdocuments#1335]
**Describe the bug**
After installing Collabora app Nextcloud generates previews for Office and PDF files despite that it is not turned on in 'enabledPreviewProvieders' in nextcloud config.

**To Reproduce**
1. Do not turn on or disable in config.php Office and PDF preview generation.
2. Install Collabora app
3. Create new Office or PDF document - preview is generated.
4. Do the same with Collabora app turned off - preview is not generated as expected.

**Expected behavior**
It should not generate previews when not configured. According to Nextcloud documentation:

```
The following providers are disabled by default due to performance or privacy concerns:

        OC\Preview\Illustrator
        OC\Preview\Movie
        OC\Preview\MSOffice2003
        OC\Preview\MSOffice2007
        OC\Preview\MSOfficeDoc
        OC\Preview\OpenDocument
        OC\Preview\PDF
        OC\Preview\Photoshop
        OC\Preview\Postscript
        OC\Preview\StarOffice
        OC\Preview\SVG
        OC\Preview\TIFF
        OC\Preview\Font
```

**Screenshots**

**Client details:**
 - Device: [e.g. iPhone6, desktop]

## Server details

**Operating system**: Ubuntu 18.04 LTS

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.3

**Nextcloud version:** 20.0.6

**Version of the richdocuments app** 3.7.13

**Version of Collabora Online** 6.4

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```

</details>

**Comment by juliushaertl** [comment]
Yeah the app registered preview providers are added in addition to the ones provided in the config, I assume we could have a setting to turn preview generation off. In case you are running a real Collabora server you could just block the Nextcloud server from using the `<net><post_allow>` settings

**Comment by cm-schl** [comment]
Hi,

I’ve found many .tmp files on the server under `/tmp/systemd-private--apache2.service/tmp/` that were never cleaned up by the system. Looking into those files I saw, that these are actually big .pdf files with the extension .tmp. As these were never deleted I’ve looked into the logs and there shows up an error `“message”:“Failed to convert file to preview”` when executing `/index.php/core/preview?fileId=12345&x=32&y=32`. This seems to be connected to a timeout when getting the file to generate the preview.

I haven’t been successful in finding the source of the timeout and I didn't wanted to have a preview of PDFs. But I can confirm that the same problem happens on my server: richdocuments generates previews (or tries it) of PDF files. To reduce the risk of running out of space on the server I modified the file [Office.php ](https://github.com/nextcloud/richdocuments/blob/master/lib/Preview/Office.php) so that it never creates previews.

I think the best option would be to look into the configuration in config.php as users do not expect that there is a second configuration option for previews. This somehow would also solve https://github.com/nextcloud/richdocuments/issues/461 .

**Comment by rainerjung** [comment]
Agreed, the preview generation is a lot of overhead and times out for big documents. Making it configurable or even better respecting the existig server config would be great. Thanks!

**Comment by Ramalama2** [comment]
It gots urgent i think, because previewgenerator with imaginary in combination with collabora, is a pain, everytime you modify the Document, the file gets Locked because of previewgenerator and you get an error message that pops up, that saving the file was not possible in collabora.

Its basically unusable. Im not sure how to fix that. Either Collabora needs an option or Previegenerator needs an disabledPreviewProviders function.

**Comment by fcharlaix-opendsi** [comment]
Hi,

I have encountered the same behavior, the `enabledPreviewProviders` seems correctly set : 
```php
root@vm1srv-cloudweb:~# sudo -u www-data php /var/www/nextcloud/occ config:system:get enabledPreviewProviders
OC\Preview\BMP
OC\Preview\GIF
OC\Preview\JPEG
OC\Preview\MarkDown
OC\Preview\MP3
OC\Preview\PNG
OC\Preview\TXT
OC\Preview\XBitmap
OC\Preview\Krita
```

But the Collabora server still receives `POST /cool/convert-to/png` from the Nextcloud server and the `preview` folder take around 72G.

**Comment by juliusknorr** [comment]
https://github.com/nextcloud/richdocuments/pull/4426 added a config option now
